src/pages/costModels/costModel/addSourceWizard.tsx

```diff
@@ -49,7 +50,8 @@ const sourceTypeMap = {
 };
 
 class AddSourceWizardBase extends React.Component<Props, AddSourcesStepState> {
-  public state = { checked: {} };
+  private checkChecked = (a, b) => Object.keys(a).every(key => b[key] && a[key].selected === b[key].selected);
```

Comment:
This will not work AS if there is a preexisting item(source) that has been assign to the cost model it still shows up in the list and it shows up as selected(true). So if we test to see if an item is selected(true) and enable the button it would always be enable for any cost model that already has a source assigned to it. This is not what we want. Taking the initial value and testing it against any changes that vary from that initial value captures all test cases.  In this case we are just cycling cycling through the state objects and comparing the the "selected" lists to see if they differ.

Reply:
I am testing a new approach this morning, it still involves having an initial state as we need that, but what i am thinking is that I can capture the initial total of selected items. then we can easily have something like this

`    If (selectedCount === initialSelectedCount) {
        disableButton = true;
    }`

Reply:
I pushed back changes using counts, this should cleanup some of the comparison calls.

Reply:
Sorry if I wasn't being clear, but we can still reuse the existing `checked` state. The addSourceStep component shows the initial checked state as disabled. We simply need to check that as well, then look for at least one `selected true` item.

See https://github.com/project-koku/koku-ui/pull/2073/files

Reply:
OK looked at what you had, I switched up the code base to follow that. I have to admit I missed the fact that we were already tracking the disabled in the step class.